docs/user_guides/fs/feature_group/create.md

```diff
@@ -101,6 +78,178 @@ MaxDirectoryItemsExceededException - The directory item limit is exceeded: limit
+    even with pushed down filters.
+
+    A good practice are partition keys with at most daily granularity, if they are based on time.
+
+Additionally, if you are commonly training models for different categories of your data, you can add another level of partitioning for this. That is, if the query contains
+an additional filter:
+```python
+query = fg.select_all().filter(fg.country_code == "US")
+```
+
+The feature group can be created with the following partition key in order to push down filters also for the `country_code` category:
+```python
+fg = feature_store.create_feature_group(...
+    partition_key=['day', 'country_code'],
+    event_time='day',
+)
+```
+
+##### Parquet file size within a feature group partition
+
+Once you have decided on the feature group level partitioning and you start inserting data to the feature group, there are multiple ways in order to
+influence how Hudi will **split the data between parquet files within the feature group partitions**.
+The two things that influence the number of parquet files per partition are
+
+1. The number of feature group partitions written in a single insert
+2. The shuffle parallelism used by Hudi
+
+In general, the inserted dataframe (unique combination of partition key values) will be parallised according to the following Hudi settings:
+!!! example "Default Hudi partitioning"
+    ```python
+    write_options = {
+        'hoodie.bulkinsert.shuffle.parallelism': 5,
+        'hoodie.insert.shuffle.parallelism': 5,
+        'hoodie.upsert.shuffle.parallelism': 5
+    }
+    ```
+That means, using Spark, Hudi shuffles the data into five in-memory partitions, which each fill map to a task and finally a parquet file (see figure below).
+If the inserted Dataframe contains only a single feature group partition, this feature group partition will be written with five parquet files.
+If the inserted Dataframe contains multiple feature group partitions, the parquet files will be split among those partition, potentially more parquet files will be added.
+
+<figure markdown>
+  ![feature group partitioning](../../../assets/images/guides/feature_group/fg-partition-files.png)
+  <figcaption>Mapping in-memory partitions to tasks, workers, executors and feature group partition files for a feature group insert</figcaption>
+</figure>
+
+!!! tip "Setting shuffle parallelism"
+    In practice that means the shuffle parallelism should be set equal to the number of feature group partitions being contained in the inserted dataframe.
+    This will create one parquet file per feature group partition, which in many cases is optimal.
+
+    Theoretically, this rule holds up to a partition size of 2GB, which is the limit of Spark. However, one should bump this up accordingly already for smaller inputs.
+    We recommend having shuffle parallelism `hoodie.[insert|upsert|bulkinsert].shuffle.parallelism` such that its at least input_data_size/500MB.
```

Comment:
I wonder how this interacts with the `write.parquet.max.file.size` property (https://hudi.apache.org/docs/configurations/#writeparquetmaxfilesize)